cfpb/sbl-frontend
# [DRAFT] feat(deps): update to node 18, yarn 4

Found this bug when trying to run tests locally:

<img width="961" alt="Screenshot 2024-04-27 at 4 43 24 PM" src="https://github.com/cfpb/sbl-frontend/assets/19983248/669e9a58-a713-435c-a5b9-0c131b46938d">

This bug is [caused by](https://github.com/yarnpkg/berry/pull/6030) running on an old version of node on older versions of yarn when using vitest, which has [been fixed in later updates](https://github.com/yarnpkg/berry/pull/6030). Updating to the latest version of yarn fixes this issue (the [DS did it earlier this year](https://github.com/cfpb/design-system/commit/2691a16d1ec50bf9dcac6aca63fab19573742f1f)), but v4 of yarn also requires at least node 18. We should at least update from node 16 to node 18 anyway for the engine, [since it's at EOL](https://github.com/nodejs/Release). We could also update all the way to node 20, or even 22 if we're ok with it not being a LTS release for a few months. We can probably wait till post-mvp for that change.

## Changes
- update yarn to v4
- update node to 18, maybe even 20/22 post-mvp if we're ok not being on a LTS release for a few months

## How to test this PR

- make sure you're using node 18, if you're using nvm, it's `nvm use 18`
- make sure you're using the prescribed version of yarn, it's `yarn set version 4.1.1` or use corepack
- does vite still build things correctly? `yarn run dev`
- can you run your tests locally? `yarn test`

## Screenshots

### Before

<img width="961" alt="Screenshot 2024-04-27 at 4 43 24 PM" src="https://github.com/cfpb/sbl-frontend/assets/19983248/1f0c7dcf-9419-4961-a6d6-8b5754fc8300">

### After
![Screenshot 2024-04-28 at 9 34 36 AM](https://github.com/cfpb/sbl-frontend/assets/19983248/d9078f72-1586-4170-a738-b3030da9816a)
